dandi/bids_validator_xs.py

```diff
@@ -581,31 +604,28 @@ def select_schema_dir(
             else:
                 with open(dataset_description) as f:
                     dataset_info = json.load(f)
-                    if force_select:
-                        try:
-                            schema_version = dataset_info["BIDSVersion"]
-                        except KeyError:
-                            return utils.get_schema_path()
-                    else:
+                    try:
                         schema_version = dataset_info["BIDSVersion"]
+                    except KeyError:
+                        schema_version = schema_min_version
```

Comment:
above can be oneliner
```
schema_version = dataset_info.get("BIDSVersion", schema_min_version)
```
unless you may be want to warn the user that dataset at that path lacks BIDSVersion which is REQUIRED.  So -- addd here a `lgr.warning` stating that this dataset_description lacks required field and you are choosing the schema_min_version for that

Reply:
Makes sense, the actual complaint about REQUIRED might best come out in the report — i.e. report the file is broken in the same place where we would report all broken files, rather than report here as a special case.